[linux-6.6.y] Turning off Zhaoxin ahci controller runtime pm from sysfs

leoliu-oc opened the pull request:
zhaoxin inclusion
category: bugfix

-------------------

There exits some problems with enabling ahci controller runtime pm on various Zhaoxin platforms.

For some Zhaoxin CPUs, link into listen mode, and PHY into P2, which will cause the device to continuously send comminit. When the controller is in D3, the comminit will continuously wake up the controller, resulting in the controller being unable to stabilize at D3;

For another Zhaoxin CPUs, after entering and exiting runtime pm for a certain number of times, ahci controller reset accumulates for many times, which will make ahci and PXPTRF P2CW think that the other party can no longer receive requests and block the P2CW path, resulting in failure to recognize the device;

There are alse some Zhaoxin CPUs, the test is normal, but considering that the server platform is not sensitive to power consumption, and this part has very little impact on power consumption;

In summary, it is recommended to close the zx ahci controller runtime pm.

opsiff commented on drivers/base/power/sysfs.c:
```diff
@@ -108,7 +109,19 @@ static ssize_t control_show(struct device *dev, struct device_attribute *attr,
+	    pdev->device == 0x9083 &&
+	    pdev->revision <= 0x20) {
+		device_unlock(dev);
+		return -EPERM;
```
这个EPERM值不合理 应该是EINVAL

leoliu-oc replied:
这个补丁有两个方案，根据建议这边修改为了另一个方案。还烦请review了。谢谢。

opsiff commented:
而且不应该在sysfs.c这个路径上判断 应该在具体驱动里面改

leoliu-oc committed:
Turning off Zhaoxin ahci controller runtime pm from sysfs

zhaoxin inclusion
category: bugfix

-------------------

There exits some problems with enabling ahci controller runtime pm on
various Zhaoxin platforms.

For some Zhaoxin CPUs, link into listen mode, and PHY into P2, which will
cause the device to continuously send comminit. When the controller is in
D3, the comminit will continuously wake up the controller, resulting in
the controller being unable to stabilize at D3;

For another Zhaoxin CPUs, after entering and exiting runtime pm for a
certain number of times, ahci controller reset accumulates for many times,
which will make ahci and PXPTRF P2CW think that the other party can no
longer receive requests and block the P2CW path, resulting in failure to
recognize the device;

There are alse some Zhaoxin CPUs, the test is normal, but considering that
the server platform is not sensitive to power consumption, and this part
has very little impact on power consumption;

In summary, it is recommended to close the zx ahci controller runtime pm.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

leoliu-oc commented:
> 而且不应该在sysfs.c这个路径上判断 应该在具体驱动里面改

新版本中已经修改了，不是在sysfs.c中。
